stdlib/universe/state_tracking.go

```diff
@@ -186,14 +183,13 @@ func (t *stateTrackingTransformation) Process(id execute.DatasetID, tbl flux.Tab
 	if !created {
 		return errors.Newf(codes.FailedPrecondition, "found duplicate table with key: %v", tbl.Key())
 	}
-	err := execute.AddTableCols(tbl, builder)
-	if err != nil {
+	if err := execute.AddTableCols(tbl, builder); err != nil {
 		return err
 	}
 
 	// Prepare the functions for the column types.
 	cols := tbl.Cols()
-	err = t.fn.Prepare(cols)
+	fn, err := t.fn.Prepare(cols)
```

Comment:
Oh, I think I see now. You've moved the error collection to `prepare` and taken it off `dynamic`.

Reply:
Oh, nvm. You're just no longer returning the state from the `dynamicFn` since you're not keeping it there anymore. Forgive me. Big diff.